lib/request.js

```diff
@@ -1,18 +1,28 @@
 const http = require('http');
 const https = require('https');
+const zlib = require('zlib');
 
 module.exports = (options) => {
     options = Object.assign({
         protocol: 'https:'
     }, options);
     const httpModule = options.protocol === 'https:' ? https : http;
     return new Promise((resolve, reject) => {
-        const chunks = [];
         const request = httpModule.request(options, (response) => {
-            response.on('data', (chunk) => {
+            let bodyStream;
+            const chunks = [];
+            const encoding = response.headers && response.headers['content-encoding'];
+            if (encoding === 'gzip' || encoding === 'deflate') {
+                response.on('error', reject);
+                bodyStream = response.pipe(zlib.createUnzip());
+            } else {
+                bodyStream = response;
+            }
+            bodyStream.on('error', reject);
```

Comment:
@shuhei do you think that it's enough? I mean probably it's better to make some cleanup, removing handlers, chunks array and so on. if error occurred and promise will be rejected original stream objects will be still in place..